packages/core/src/components/Popover/index.tsx

```diff
@@ -0,0 +1,308 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import React from 'react';
+import { v4 as uuid } from 'uuid';
+import Text from '../Text';
+import withStyles, { WithStylesProps } from '../../composers/withStyles';
+import { styleSheetPopover } from './styles';
+import Portal from '../Portal';
+const EMPTY_TARGET_RECT: ClientRect = {
+  bottom: 0,
+  height: 0,
+  left: 0,
+  right: 0,
+  top: 0,
+  width: 0,
+};
+export type PopoverProps = {
+  /** Accessibility label. If not specified, all popover content is duplicated, rendered in an off-screen element with a separate layer. */
+  accessibilityLabel?: string;
+  /** Inline content to hover. */
+  children: NonNullable<React.ReactNode>;
+  /** What to show in the popover. */
+  content: NonNullable<React.ReactNode>;
+  /** True to disable popover but still show children. */
+  disabled?: boolean;
+  /** Manually override calculated horizontal align */
+  horizontalAlign?: 'center' | 'left' | 'right';
+  /** True to use a light background with dark text. */
+  inverted?: boolean;
+  /** Number of seconds the cursor must hover over the trigger to trigger the popoover */
+  mouseEnterDelay?: number;
+  /** Number of seconds popover will persist if you do not move your cursor into the popover. */
+  mouseLeaveDelay?: number;
+  /** Callback fired when the popover is shown. */
+  onShow?: () => void;
+  /** True to prevent dismissmal on mouse down. */
+  remainOnMouseDown?: boolean;
+  /** True to toggle popover display on click.  */
+  toggleOnClick?: boolean;
+  /** True to add a dotted bottom border. */
+  underlined?: boolean;
+  /** Manually override calculated vertical align */
+  verticalAlign?: 'above' | 'below';
+  /** Width of the popover in units. */
+  width?: number;
+};
+export type PopoverState = {
```

Comment:
please use new lines between blocks of code (between type blocks, between functions, etc.), so it's easier to read and not squished.